src/collect/output.rs

```diff
@@ -0,0 +1,103 @@
+use std::{
+    fs::OpenOptions,
+    io::{self, BufWriter, Write},
+};
+
+use anyhow::Result;
+
+use super::cli::CollectArgs;
+use crate::core::events::Event;
+
+/// Given a Formatter and cli arguments, get a list of output processors, for
+/// later event output processing.
+pub(super) fn get_processors<'a, F: Formatter>(
+    formatter: &'a mut F,
+    args: &CollectArgs,
+) -> Result<Vec<Box<dyn Processor + 'a>>> {
+    // The actual output selection will be decided by the user (--out and/or
+    // --print) below.
+    let mut writers: Vec<Box<dyn Write>> = Vec::new();
+
+    // Write the events to a file if asked to.
+    if let Some(out) = args.out.as_ref() {
+        writers.push(Box::new(BufWriter::new(
+            OpenOptions::new()
+                .create(true)
+                .write(true)
+                .truncate(true)
+                .open(out)?,
```

Comment:
Maybe we could wrap the errors that can happen here and print the file that we're trying to open.
RIght now this is the first file we handle but when we have profiles, only printing "Not such file or directory" will be confusing.